Source/items.cpp

```diff
@@ -420,42 +420,47 @@ void InitItemGFX()
 	memset(UniqueItemFlags, 0, sizeof(UniqueItemFlags));
 }
 
-bool ItemPlace(int xp, int yp)
+bool ItemPlace(Point position)
 {
-	if (dMonster[xp][yp] != 0)
+	if (dMonster[position.x][position.y] != 0)
 		return false;
-	if (dPlayer[xp][yp] != 0)
+	if (dPlayer[position.x][position.y] != 0)
 		return false;
-	if (dItem[xp][yp] != 0)
+	if (dItem[position.x][position.y] != 0)
 		return false;
-	if (dObject[xp][yp] != 0)
+	if (dObject[position.x][position.y] != 0)
 		return false;
-	if ((dFlags[xp][yp] & BFLAG_POPULATED) != 0)
+	if ((dFlags[position.x][position.y] & BFLAG_POPULATED) != 0)
 		return false;
-	if (nSolidTable[dPiece[xp][yp]])
+	if (nSolidTable[dPiece[position.x][position.y]])
 		return false;
 
 	return true;
 }
 
+Point GetAvailableItemPosition()
+{
+	Point position = {};
+	do {
+		position = Point { GenerateRnd(80), GenerateRnd(80) } + Point { 16, 16 };

```

Comment:
Since the 16 represents the level boarder (levels are 112x112 where only the inner 80x80 tiles are playable this looks more readable to me:
```suggestion
		position = Point { 16 + GenerateRnd(80), 16 + GenerateRnd(80) };
```

Reply:
I think this is arbitrary. Why do you think it's more readable? It's the opposite to me: the "{ 16, 16 }" represents the border as a single entity instead of being spread out as separate components in each coordinate. It even helps with extra refactoring later to isolate it as a "border" member somewhere.

Reply:
The part i like is that it's first since we are offsetting top and left, not rigth and bottom.

So this would work equally well:
```suggestion
		position = Point { 16, 16 } + Point { GenerateRnd(80), GenerateRnd(80) };
```

Reply:
That sounds sensible. Will change.